firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.kt

```diff
@@ -0,0 +1,383 @@
+     * providerInstalled lock.
+     */
+    private var providerInstallStarted = false
+
+    /**
+     * Runs ProviderInstaller.installIfNeededAsync once per application instance.
+     *
+     * @param context The application context.
+     * @param uiExecutor
+     */
+    private fun maybeInstallProviders(context: Context, uiExecutor: Executor) {
+      // Make sure this only runs once.
+      synchronized(providerInstalled) {
+        if (providerInstallStarted) {
+          return
+        }
+        providerInstallStarted = true
+      }
+
+      // Package installIfNeededAsync into a Runnable so it can be run on the main thread.
+      // installIfNeededAsync checks to make sure it is on the main thread, and throws otherwise.
+      uiExecutor.execute {
+        ProviderInstaller.installIfNeededAsync(
+                context,
+                object : ProviderInstaller.ProviderInstallListener {
+                  override fun onProviderInstalled() {
+                    providerInstalled.setResult(null)
+                  }
+
+                  override fun onProviderInstallFailed(i: Int, intent: Intent?) {
+                    Log.d("FirebaseFunctions", "Failed to update ssl context")
+                    providerInstalled.setResult(null)
+                  }
+                })
+      }
+    }
+
+    /**
+     * Creates a Cloud Functions client with the given app and region or custom domain.
+     *
+     * @param app The app for the Firebase project.
+     * @param regionOrCustomDomain The region or custom domain for the HTTPS trigger, such as `"us-central1"` or `"https://mydomain.com"`.
+     */
+    @JvmStatic
+    fun getInstance(
+            app: FirebaseApp, regionOrCustomDomain: String): FirebaseFunctions {
+      Preconditions.checkNotNull(app, "You must call FirebaseApp.initializeApp first.")
+      Preconditions.checkNotNull(regionOrCustomDomain)
+      val component = app.get(FunctionsMultiResourceComponent::class.java)
+      Preconditions.checkNotNull(component, "Functions component does not exist.")
+      return component[regionOrCustomDomain]
+    }
+
+    /**
+     * Creates a Cloud Functions client with the given app.
+     *
+     * @param app The app for the Firebase project.
+     */
+    @JvmStatic
+    fun getInstance(app: FirebaseApp): FirebaseFunctions {
+      return getInstance(app, "us-central1")
+    }
+
+    /**
+     * Creates a Cloud Functions client with the default app and given region or custom domain.
+     *
+     * @param regionOrCustomDomain The region or custom domain for the HTTPS trigger, such as `"us-central1"` or `"https://mydomain.com"`.
+     */
+    @JvmStatic
+    fun getInstance(regionOrCustomDomain: String): FirebaseFunctions {
+      return getInstance(FirebaseApp.getInstance(), regionOrCustomDomain)
+    }
+
+    val instance: FirebaseFunctions
+      /** Creates a Cloud Functions client with the default app.  */
+      get() = getInstance(FirebaseApp.getInstance(), "us-central1")
```

Comment:
The converter removed a getter and also created a field just for a getter here.

Reply:
yeah- seems like getters/setters are gonna be a moderate hurdle